lib/osf-components/addon/components/resources-list/edit-resource/component.ts

```diff
@@ -0,0 +1,94 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+// import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { ValidationObject } from 'ember-changeset-validations';
+import { validatePresence } from 'ember-changeset-validations/validators';
+import { task } from 'ember-concurrency';
+// import { taskFor } from 'ember-concurrency-ts';
+import DS from 'ember-data';
+import IntlService from 'ember-intl/services/intl';
+import RegistrationModel from 'ember-osf-web/models/registration';
+import ResourceModel, { ResourceTypes } from 'ember-osf-web/models/resource';
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
+import { tracked } from 'tracked-built-ins';
+// import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+
+interface Args {
+    resource?: ResourceModel;
+    registration?: RegistrationModel;
+    reload?: () => {};
+}
+
+interface ResourceValidations {
+    pid: string;
+    description: string;
+    resourceType: string;
+}
+
+export default class EditResourceModal extends Component<Args> {
+    @service toast!: Toastr;
+    @service intl!: IntlService;
+    @service store!: DS.Store;
+
+    availableTypes = Object.values(ResourceTypes);
+    resourceValidations: ValidationObject<ResourceValidations> = {
+        pid: [validatePresence({
+            type: 'invalid',
+            presence: true,
+            translationArgs: { description: this.intl.t('osf-components.resources-list.edit_resource.doi') },
+        })],
+        resourceType: [validatePresence({
+            type: 'blank',
+            presence: true,
+            translationArgs: { description: this.intl.t('osf-components.resources-list.edit_resource.output_type') },
+        })],
+        description: [validatePresence({
+            type: 'blank',
+            presence: true,
+            translationArgs: { description: this.intl.t('osf-components.resources-list.edit_resource.description') },
+        })],
+    };
+
```

Comment:
Shouldn't there be a `title` field as well?

Reply:
There isn't in the mockup though. The `resource-card` doesn't show title as well.

Reply:
Hmm. Might be something to discuss at scrum. I think both Jon and I thought there would be, but yeah, maybe not.